src/main/java/org/apache/maven/shared/dependency/analyzer/DefaultProjectDependencyAnalyzer.java

```diff
@@ -276,12 +254,7 @@ else if ( file != null && file.isDirectory() )
 
             if ( artifact != null )
             {
-                Set<String> classesFromArtifact = usedArtifacts.get( artifact );
-                if ( classesFromArtifact == null )
-                {
-                    classesFromArtifact = new HashSet<String>();
-                    usedArtifacts.put( artifact, classesFromArtifact );
-                }
+                Set<String> classesFromArtifact = usedArtifacts.computeIfAbsent( artifact, k -> new HashSet<>() );
```

Comment:
This one's more borderline, though I still find the old version clearer. 

That said, is anything actually changing here? If this isn't relevant to fixing the issue at hand, then we should try to avoid unnecessary churn that obscures what's actually changing in this PR.